torchtext/experimental/datasets/text_classification.py

```diff
@@ -42,6 +43,7 @@ def _generate_data_iterators(dataset_name, root, ngrams, tokenizer, data_select)
 
     dataset_tar = download_from_url(URLS[dataset_name], root=root)
     extracted_files = extract_archive(dataset_tar)
+    extracted_files = [os.path.join(root, f) for f in extracted_files]
```

Comment:
This is BC breaking

Reply:
See above!

Reply:
I would prefer a separate PR to address this issue. It makes this PR a little bit overwhelming. I agree that both download/extract functions should handle zip/tar files consistently. @cpuhrsch how do you feel about it?

Reply:
Agreed. It's unrelated to this and we might want to explicitly communicate this change.

Reply:
Resolved by #692